### PR TITLE
Updated protobufs, expanded supported types

### PR DESCRIPTION
**Pull request by katarzyna-z**
Summary of changes:
- Updated protobufs, 
- Expanded supported types

### CODE REVIEW

File: src/snap/metric.cc [hunk 1]
```diff
@@ -199,6 +217,10 @@ int Metric::get_int_data() const {
   return rpc_metric_ptr->int32_data();
 }
 
+long int Metric::get_int64_data() const {
```
Comment (tomaszkapela) [inline src/snap/metric.cc]: Does `int64_data()` throw? It's generally good to mark functions that don't throw as `noexcept` for performance reasons. It is an API contract though, so if you're not sure, better not add it.

File: src/snap/metric.cc [hunk 2]
```diff
@@ -190,6 +188,26 @@ void Metric::set_data(int data) {
   rpc_metric_ptr->set_int32_data(data);
 }
 
+void Metric::set_data(long int data) {
+  type = DataType::Int64;
```
Comment (aarontay) [inline src/snap/metric.cc]: if compiling in 32 bit mode, long would only be int32--I would suggest updating to be explicit and use int32_t, int64_t, uint32_t, uint64_t as well.

**Comment by aarontay** [comment]
It might be worth updating 'example/publisher/src/log.cc' to handle all types as well to provide a template plugin for users on the expanded types: 
```
    switch (mets_iter->data_type()) {
      case Metric::DataType::String:
        outfile << mets_iter->get_float64_data() << "\n";
        break;
      case Metric::DataType::Float32:
        outfile << mets_iter->get_float32_data() << "\n";
        break;
      case Metric::DataType::Float64:
        outfile << mets_iter->get_float64_data() << "\n";
        break;
      case Metric::DataType::Int32:
        outfile << mets_iter->get_int_data() << "\n";
        break;
      case Metric::DataType::Int64:
        outfile << mets_iter->get_int_data() << "\n";
        break;
      case Metric::DataType::Uint32:
        outfile << mets_iter->get_int_data() << "\n";
        break;
      case Metric::DataType::Uint64:
        outfile << mets_iter->get_int_data() << "\n";
        break;
      case Metric::DataType::Bool:
        outfile << mets_iter->get_int_data() << "\n";
        break;
      case Metric::DataType::NotSet:
        outfile << "not set\n";
        break;
```

File: src/snap/metric.cc [hunk 3]
```diff
@@ -15,10 +15,12 @@ limitations under the License.
 
 #include <utility>
 #include <chrono>
+#include <cstdint>
```
Comment (tomaszkapela) [inline src/snap/metric.cc]: The header file with the declaration of these functions already has this include, I don't see a need to put it here as well. It's the same with the other includes.

File: examples/collector/src/rando.cc [hunk 4]
```diff
@@ -26,6 +28,29 @@ using Plugin::Metric;
...
+enum supported_types
+{
+    float32,
+    float64,
+    int32,
+    int64,
+    uint32,
+    uint64,
+    boolean,
+	string
```
Comment (tomaszkapela) [inline examples/collector/src/rando.cc]: Something's wrong with the indentation.

**Comment by katarzyna-z** [comment]
@tomaszkapela thanks for review! :tulip: 
@aarontay please take a look once again;)

**Pull request merged**